File: 30-Days-of-Code/R/day2_operators.R
# Enter your code here. Read input from STDIN. Print output to STDOUT

vec <- readLines(file("stdin"))

meal_cost <- as.numeric(vec[1])
tip_percent <- as.integer(vec[2])
tax_percent <- as.integer(vec[3])

tip_percent <- tip_percent / 100
tax_percent <- tax_percent / 100 

tip <- meal_cost * tip_percent
tax <- meal_cost * tax_percent

total_cost <- round(meal_cost + tip + tax)

write(paste("The total meal cost is", total_cost, "dollars."), stdout())

File: 30-Days-of-Code/Java/day2_operators.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        double meal_cost = in.nextDouble();
        int tip_percent = in.nextInt();
        int tax_percent = in.nextInt();
        in.close();
        System.out.println("The total meal cost is " + totalCost(meal_cost, tip_percent, tax_percent) + " dollars.");
    }

    public static int totalCost(double mealCost, int tipPercent, int taxPercent) {
        double tip_temp = (double) tipPercent/100;
        double tax_temp = (double) taxPercent/100;
        double tip = mealCost * tip_temp;
        double tax = mealCost * tax_temp;
        int result = (int) Math.round(mealCost + tip + tax);
        return result;
    }
}
